lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt

```diff
@@ -103,10 +175,35 @@ interface Account {
         calldata: Calldata,
         salt: Felt,
         maxFee: Felt,
-    ): DeployAccountTransactionPayload {
+    ): DeployAccountTransactionV1Payload {
         return signDeployAccount(classHash, calldata, salt, maxFee, Felt.ZERO, false)
     }
 
+    /**
+     * Sign version 3 deploy account transaction.
+     *
+     * Sign a deploy account transaction that requires prefunding deployed address.
+     *
+     * @param classHash hash of the contract that will be deployed. Has to be declared first!
+     * @param calldata constructor calldata for the contract deployment
+     * @param salt salt used to calculate address of the new contract
+     * @param forFeeEstimate when set to `true`, it changes the version to `2^128+version` so the signed transaction can only be used for fee estimation
+     * @return signed deploy account payload
+     */
+    fun signDeployAccount(
+        classHash: Felt,
+        calldata: Calldata,
+        salt: Felt,
+        l1ResourceBounds: ResourceBounds,
+        forFeeEstimate: Boolean = false,
+    ): DeployAccountTransactionV3Payload {
+        val params = DeployAccountParamsV3(
+            nonce = Felt.ZERO,
```

Comment:
are you sure it is safe to set nonce to 0 here?

Reply:
Yes, as it was done previously in v1 deploy account. 
Nonce cannot be increased without deployed account as far as I am aware.